### PR TITLE
Split up `Books` component into separate files

### PR DESCRIPTION
**Pull request by drewwyatt**
This could benefit a lot from `useState` and `useEffect` (seen [here](https://github.com/daltonclaybrook/dalton-website/pull/2/files#diff-997a3ae09b240a8fcfd8de4a1ec7fa8dR1)), but don't let that detract from some of the organizational changes here.

Even without hooks, the Books component could benefit from splitting out some of the api/view concerns and you would end up with something like:

```typescript
class Books extends Component<{}, IBooksState> {
  constructor(props: {}) {
    super(props);
    this.state = {
        books: [],
    };
  }

  public componentDidMount() {
      this.fetchBooks();
  }

  public render() {
    return <View books={this.state.books} />;
  }

  // MARK: - Helpers

  private async fetchBooks() {
    const books = await fetchBooks();
    this.setState({ books });
  }
}
```

### CODE REVIEW

File: src/components/Books/Books.tsx [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+import IBook from '../../models/IBook';
+import { fetchBooks } from './api';
+import View from './BooksView';
+
+interface IBooksState {
+    books: IBook[];
+}
+
+const Books: SFC = () => {
+  const [books, loadBooks] = useState<IBook[]>([]);
```
Comment (drewwyatt) [inline src/components/Books/Books.tsx]: `useState` takes the default value for the state you are looking to create (`[]`, in this case), and returns a tuple with the current state (`books`) and a state update method (`loadBooks`).

File: src/components/Books/Books.tsx [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  useEffect(() => {
```
Comment (drewwyatt) [inline src/components/Books/Books.tsx]: This actually works without wrapping the api call in `useEffect`, but that still seems to be the recommended way to handle this. From [the docs](https://reactjs.org/docs/hooks-effect.html):

> Tip
> If you’re familiar with React class lifecycle methods, you can think of useEffect Hook as componentDidMount, componentDidUpdate, and componentWillUnmount combined.

File: package.json [hunk 3]
```diff
@@ -7,8 +7,8 @@
     "@types/node": "^10.12.18",
     "@types/react": "^16.7.18",
     "@types/react-dom": "^16.0.11",
-    "react": "^16.7.0",
-    "react-dom": "^16.7.0",
+    "react": "^16.7.0-alpha.2",
```
Comment (drewwyatt) [inline package.json]: Using an alpha version might be a bit spooky. I totally get it if you don't wanna do this - which is why I provided a non-hooks example in the description.

File: src/components/Books/Books.tsx [hunk 4]
```diff
@@ -0,0 +1,18 @@
+import React, { SFC, useEffect, useState } from 'react';
+import IBook from '../../models/IBook';
+import { fetchBooks } from './api';
+import View from './BooksView';
```
Comment (daltonclaybrook) [inline src/components/Books/Books.tsx]: I was pretty interested to see what you would recommend here with my `Books` vs `Book`. I like this.

**Pull request merged**